### PR TITLE
Phase 4

### PR DESCRIPTION
**Pull request by adorjan**
Spring MVC + JSP + Taglib

### CODE REVIEW

File: pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class PancakeOrderValidator {
+    private PancakeOrderValidator() {
+        super();
+    }
+
+    public Map<Pancake, String> getValidPancakeOrder(Map<Pancake, String> pancakesWithAmounts) {
+        Map<Pancake, String> pancakesWithNonZeroAmounts = new HashMap<Pancake, String>();
+        for (Pancake pancake : pancakesWithAmounts.keySet()) {
+            String pancakeAmount = pancakesWithAmounts.get(pancake);
+            if (pancakeAmount != null && !"".equals(pancakeAmount) && !"0".equals(pancakeAmount)) {
```
Comment (kicsikrumpli) [inline pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java]: !pancakeAmount.isEmpty()

Comment (kicsikrumpli) [inline pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java]: In such cases it is a good practice to extract magic strings as constants

File: pancake-webapp/src/main/webapp/WEB-INF/spring-core-config.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<beans xmlns="http://www.springframework.org/schema/beans"
+    xmlns:context="http://www.springframework.org/schema/context"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:mvc="http://www.springframework.org/schema/mvc"
```
Comment (kicsikrumpli) [inline pancake-webapp/src/main/webapp/WEB-INF/spring-core-config.xml]: The mvc namespace is not used

File: pancake-webapp/src/test/java/io/github/pancake/web/PancakeControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        // WHEN
+        underTest.order(mockModelAndView);
+        // THEN
+        verify(mockModelAndView, atLeastOnce()).addObject("pancakes", pancakes);
+    }
+
+    @Test
+    public void testOrderConfirmationShouldAddOrderedPancakesToModelWhenInvoked() {
+        // GIVEN in setUp
+        Mockito.when(mockPancakeOrderValidator.getValidPancakeOrder(orderedPancakes)).thenReturn(orderedPancakes);
```
Comment (kicsikrumpli) [inline pancake-webapp/src/test/java/io/github/pancake/web/PancakeControllerTest.java]: Suggest BDDMocikto - better conforms to the given-when-then naming

File: pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import io.github.pancake.persistence.base.Pancake;
+
+/**
+ * Utility class which validates the pancake order.
+ * 
+ * @author Adorjan Nagy
+ */
+public class PancakeOrderValidator {
+    private PancakeOrderValidator() {
+        super();
```
Comment (BotondSoltesz) [inline pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java]: Does _super()_ really need?

File: pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java (hunk 1 above)
Comment (BotondSoltesz) [inline pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java]: I encourage you to move _pancakeAmount != null && !"".equals(pancakeAmount) && !"0".equals(pancakeAmount)_ expression to a well named method.

File: pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package io.github.pancake.web;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.github.pancake.persistence.base.Pancake;
+
+/**
+ * Utility class which validates the pancake order.
```
Comment (BotondSoltesz) [inline pancake-webapp/src/main/java/io/github/pancake/web/PancakeOrderValidator.java]: I would expect some error message or exception (or at least a method with a _boolean_ result) in a _Validator_ class, not **filtered** list.

File: pancake-webapp/src/test/java/io/github/pancake/web/PancakeControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * Test class for {@link PancakeController}.
+ *
+ * @author Adorjan Nagy
+ */
+public class PancakeControllerTest {
+    private PancakeController underTest;
+    @Mock
+    private PancakeFacade mockPanckakeFacade;
```
Comment (BotondSoltesz) [inline pancake-webapp/src/test/java/io/github/pancake/web/PancakeControllerTest.java]: There isn't any mocking defininition of one of _mockPanckakeFacade_ methods.

**Comment by adorjan** [comment]
Closing as this time #9 were merged instead.

**Pull request closed without merging**